apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -732,6 +732,100 @@ function getGutenboardingStatus( calypsoPort ) {
 	};
 }
 
+/**
+ * Hooks the nav sidebar to change some of its button labels and behaviour.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function getNavSidebarLabels( calypsoPort ) {
+	let allPostsLabels = null;
+	let createPostLabels = null;
+
+	const { port1, port2 } = new MessageChannel();
+	calypsoPort.postMessage(
+		{
+			action: 'getNavSidebarLabels',
+			payload: {},
+		},
+		[ port2 ]
+	);
+	port1.onmessage = ( { data } ) => {
+		allPostsLabels = data.allPostsLabels;
+		createPostLabels = data.createPostLabels;
+	};
+
+	addFilter(
+		'a8c.WpcomBlockEditorNavSidebar.allPostsLabel',
+		'wpcom-block-editor/getNavSidebarLabels',
+		( label, postType ) => ( allPostsLabels && allPostsLabels[ postType ] ) || label
+	);
+
+	addFilter(
+		'a8c.WpcomBlockEditorNavSidebar.createPostLabel',
+		'wpcom-block-editor/getNavSidebarLabels',
+		( label, postType ) => ( createPostLabels && createPostLabels[ postType ] ) || label
+	);
+}
+
+/**
+ * Retrieves info to allow the bridge to build calypso urls. Hook parts of
+ * the editor that use this info.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function getCalypsoUrlInfo( calypsoPort ) {
+	let origin = null;
+	let siteSlug = null;
+
+	const { port1, port2 } = new MessageChannel();
+	calypsoPort.postMessage(
+		{
+			action: 'getCalypsoUrlInfo',
+			payload: {},
+		},
+		[ port2 ]
+	);
+	port1.onmessage = ( { data } ) => {
+		origin = data.origin;
+		siteSlug = data.siteSlug;
+	};
+
+	addFilter(
+		'a8c.WpcomBlockEditorNavSidebar.allPostsUrl',
+		'wpcom-block-editor/getSiteSlug',
+		( url, postType ) => {
+			if ( origin && siteSlug ) {
+				if ( postType === 'page' ) {
+					return `${ origin }/pages/${ siteSlug }`;
+				} else if ( postType === 'post' ) {
+					return `${ origin }/posts/${ siteSlug }`;
+				}
+			}
+
+			return url;
+		}
+	);
+
+	addFilter(
+		'a8c.WpcomBlockEditorNavSidebar.createPostUrl',
+		'wpcom-block-editor/getSiteSlug',
+		( url, postType ) => {
+			if ( origin && siteSlug && ( postType === 'page' || postType === 'post' ) ) {
```

Comment:
In Calyspo, adding a new page or post sends me off to `wp-admin`

![add new](https://user-images.githubusercontent.com/6458278/82880414-bc7b0d80-9f81-11ea-8cf4-534ceea0933f.gif)

Reply:
> Start up dev environment for calypso, wpcom-block-editor and FSE! 3 seperate apps to make some buttons 🤦

Unfortunately this needs to be tested against `calypso.localhost:3000` because this contains changes to the calypsoify iframe. Glad the sidebar is resilient to the new calypso code not being there though! It's falling back to the dotorg site behaviour as expected.

Reply:
Argh, dang. Thanks for highlighting that. 🤦

Reply:
I also wanted to add that the buttons work and look as expected otherwise 👍